core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -73,15 +68,13 @@
 /**
  * Context object used to create lucene queries on the shard level.
  */
-public class QueryShardContext {
+public class QueryShardContext extends QueryRewriteContext {
```

Comment:
is it necessary that `QueryShardContext` extends `QueryRewriteContext`. I see a lot of parse stuff around but at this point this context shouldn't need to parse anything anymore? Sure I am missing something :)